addon/serializers/json-api-serializer.js

```diff
@@ -465,6 +472,9 @@ const JSONAPISerializer = JSONSerializer.extend({
 
 Ember.runInDebug(function() {
   JSONAPISerializer.reopen({
+    warnMessageForUndefinedType: function() {
+      return 'Encountered a resource object with an undefined type';
```

Comment:
Can we expand this error message and refer to the current serializer? Using `this.constructor.toString()` like in `warnMessageNoModelForType`